Revit_Engine/Compute/RevitDiffing.cs

```diff
@@ -155,24 +155,59 @@ private static Diff Diffing(IEnumerable<object> pastObjects, IEnumerable<object>
+            if (pastIdFragments.Count() != pastObjects.Count())
             {
                 BH.Engine.Base.Compute.RecordError($"Some of the {nameof(pastObjects)} do not have a {nameof(RevitIdentifiers)} fragment attached.");
                 return null;
             }
 
-            if (revitBHoMObjects_following.Count() != followingObjects.Count())
+            if (followingIdFragments.Count() != followingObjects.Count())
             {
                 BH.Engine.Base.Compute.RecordError($"Some of the {nameof(followingObjects)} do not have a {nameof(RevitIdentifiers)} fragment attached.");
                 return null;
             }
 
-            // Compute the diffing through DiffWithFragmentId(), which allows us to specify a Fragment and FragmentIdProperty where to find an ID to fragment by.
-            Diff revitDiff = BH.Engine.Diffing.Compute.DiffWithFragmentId(revitBHoMObjects_past, revitBHoMObjects_following, typeof(RevitIdentifiers), revitIdName, diffConfigClone);
+            List<string> pastIds, followingIds;
 
+            if (revitIdName == nameof(RevitIdentifiers.ElementId))
+            {
+                pastIds = pastIdFragments.Select(x => x.ElementId.ToString()).ToList();
+                followingIds = followingIdFragments.Select(x => x.ElementId.ToString()).ToList();
+            }
+            else
+            {
+                pastIds = pastIdFragments.Select(x => x.PersistentId.ToString()).ToList();
+                followingIds = followingIdFragments.Select(x => x.PersistentId.ToString()).ToList();
+            }
+
+            //Check for duplicate ids
+            if (pastIds.Count != pastIds.Distinct().Count())
+            {
+                string message = $"Some of the {nameof(pastObjects)} contain duplicate {revitIdName}s.";
+
+                if(pastObjects.Any(x => x.GetType().Namespace.Contains("Structure")))
+                    message+= " If trying to diff structural objects, try pulling using the Phsyical discipline rather than Structural discipline, as models containing disjointed floors and walls and/or curved beams lead to one Revit element being converted into multiple structural BHoM elements.";
+
+                BH.Engine.Base.Compute.RecordError(message);
+                return null;
+            }
+
+            if (followingIds.Count != followingIds.Distinct().Count())
+            {
+                string message = $"Some of the {nameof(followingIds)} contain duplicate {revitIdName}s.";
+
+                if (pastObjects.Any(x => x.GetType().Namespace.Contains("Structure")))
+                    message += " If trying to diff structural objects, try pulling using the Phsyical discipline rather than Structural discipline, as models containing disjointed floors and walls and/or curved beams lead to one Revit element being converted into multiple structural BHoM elements.";
+
+                BH.Engine.Base.Compute.RecordError(message);
+                return null;
+            }
```

Comment:
To avoid repeating the message contents, I thought we could rewrite this part as this:
(also there was an error on line 202: the check should've been with `followingObjects`)
```suggestion
             //Check for duplicate ids
            bool dupsInPastIds = pastIds.Count != pastIds.Distinct().Count();
            bool dupsInFollIds = followingIds.Count != followingIds.Distinct().Count();

            if (dupsInPastIds || dupsInFollIds)
            {
                List<string> messageSubjects = new List<string>();
                if (dupsInPastIds) messageSubjects.Add(nameof(pastObjects));
                if (dupsInFollIds) messageSubjects.Add(nameof(followingIds));
                string message = $"Some of the {string.Join(" and ", messageSubjects)} contain duplicate {revitIdName}s. ";

                if((dupsInPastIds && pastObjects.Any(x => x.GetType().Namespace.Contains("Structure"))) || 
                    (dupsInFollIds && followingObjects.Any(x => x.GetType().Namespace.Contains("Structure"))))
                    message += "\nIf trying to diff structural objects, try pulling using the Phsyical discipline rather than Structural discipline, as models containing disjointed floors and walls and/or curved beams lead to one Revit element being converted into multiple structural BHoM elements.";

                BH.Engine.Base.Compute.RecordError(message);
                return null;
            }
```